server/service/apple_mdm.go

```diff
@@ -4033,7 +4037,34 @@ func ReconcileAppleProfiles(
 			}
 			installTargets[p.ProfileUUID] = target
 		}
-		target.hostUUIDs = append(target.hostUUIDs, p.HostUUID)
+
+		sentToUserChannel := false
+		if p.Scope == fleet.PayloadScopeUser {
+			userEnrollment, ok := userEnrollmentMap[p.HostUUID]
+			if !ok {
+				userNanoEnrollment, err := ds.GetNanoMDMUserEnrollment(ctx, p.HostUUID)
+				if err != nil {
+					return ctxerr.Wrap(ctx, err, "getting user enrollment for host")
+				}
+				// TODO EJM We can't just error out here - need to do something different
+				if userNanoEnrollment != nil {
+					userEnrollment = userNanoEnrollment.ID
+					userEnrollmentMap[p.HostUUID] = userEnrollment
+				} else {
+					level.Warn(logger).Log("msg", "host does not have a user enrollment, falling back to system enrollment for user scoped profile",
```

Comment:
I think that's where product mentioned (in the figma) that there would be a delivery failure with "we will retry once the user channel exists". A possibly simpler approach (then marking it as failed and somehow detecting that it can be processed when a user-channel gets created) would be to leave it as pending (NULL status) so it retries on the next reconcile, until a user-channel is actually available?